rdd/RDDcounts.py

```
@@ -2,235 +2,282 @@
+        Ordered list of ontology column names to use instead of default `sample_type_groupX` columns.
+        These will be renamed internally to `<column_name><level_number>` to support level-based logic.
+    """
+
     def __init__(
         self,
-        gnps_network: str,
         sample_types: str,
-        sample_groups: List[str],
-        reference_groups: List[str],
+        gnps_network_path: Optional[str] = None,
+        sample_groups: List[str] = None,
```

Ruff (0.14.1)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)